.github/workflows/build_and_test.yml

```diff
@@ -0,0 +1,26 @@
+on: [push]
+
+name: CI
+
+jobs:
+  build_and_test:
```

Comment:
Best to have a separate build + test job. E.g. if we decide to cache the build dir, we won't have to build every push, we can just run the unit/integration tests. Can be done in the future. This is fine for now.

Reply:
Now that I think about it having a build job is pretty pointless since cargo just compiles everything when you run the tests ANYWAY so there's no point adding that step.... will get rid of it 👍